core/commands/storage/stats/stats.go

```diff
@@ -237,12 +238,14 @@ func ListHostStatsFromDatastore(ctx context.Context, node *core.IpfsNode, nodeId
 		}
 		split := strings.Split(k, "/")
 		t, err := strconv.ParseInt(split[len(split)-1], 10, 64)
-		if err != nil {
+		if err != nil || t < from || t > to {
 			continue
 		}
-		if t < from || t > to {
+		year, month, day := time.Unix(t, 0).Date()
+		if ly == year && lm == month.String() && ld == day {
```

Comment:
how can you guarantee the dates are always ordered?

Reply:
fixed